diesel-factories-code-gen/src/lib.rs

```diff
@@ -1,237 +1,368 @@
+
+        if self.is_association_field(&field.ty) {
+            None
+        } else {
+            Some(quote! {
+                #[allow(missing_docs, dead_code)]
+                pub fn #name<T: Into<#ty>>(mut self, t: T) -> Self {
+                    self.#name = t.into();
+                    self
+                }
+            })
+        }
+    }
+
+    fn gen_set_association_traits(&mut self) {
+        let association_traits = self.association_traits();
+
+        self.tokens.extend(quote! {
+            #(#association_traits)*
+        });
+    }
+
+    fn association_traits(&self) -> Vec<TokenStream> {
+        self.struct_fields()
+            .filter_map(|field| self.association_trait(field))
+            .collect()
+    }
+
+    fn association_trait(&self, field: &syn::Field) -> Option<TokenStream> {
+        use heck::CamelCase;
+
+        if self.is_association_field(&field.ty) {
+            let factory = self.factory_name();
+            let field_name = field.ident.as_ref().expect("field without name");
+            let camel_field_name = field_name.to_string().to_camel_case();
+
+            let association = self.parse_association_type(&field.ty).unwrap_or_else(|| {
+                use std::fmt::Write;
+                let mut s = String::new();
+                writeln!(
+                    s,
+                    "Invalid association attribute. Must be on one of the following forms"
+                )
+                .unwrap();
+                writeln!(s).unwrap();
+                writeln!(s, "Association<'a, Model, Factory>").unwrap();
+                writeln!(s, "Option<Association<'a, Model, Factory>>").unwrap();
+                writeln!(s, "Association<'a, Model, Factory<'a>>").unwrap();
+                writeln!(s, "Option<Association<'a, Model, Factory<'a>>>").unwrap();
+                writeln!(s).unwrap();
+                writeln!(s, "Got\n{}", self.type_to_string(&field.ty)).unwrap();
+                panic!("{}", s);
             });
 
-            if let Some(factory_attr) = maybe_factory_attr {
-                let attr = factory_attr.tts.to_string();
-                // FIXME, unicode is a valid identifier in rust!
-                let re = Regex::new(r"model = ([A-Za-z]+) ").unwrap();
-                let model_cap = re.captures(&attr).unwrap();
-                let model = &model_cap[1];
-                let model_fn = ident(&model.to_snake_case());
-                let model = ident(model);
-
-                if optional {
-                    return Some(quote! {
-                        fn #model_fn<T: diesel_factories::Association<#model_name, #model>>(mut self, association: &T) -> Self {
-                            self.#name = Some(association.id());
+            let model = association.model;
+            let other_factory = association.factory;
+
+            let other_factory_without_lifetime =
+                self.type_to_string(&other_factory).replace(" < 'a >", "");
```

Comment:
This'll fail on lifetimes that aren't named ```'a``` - probably we should be able to handle any lifetime name the user chooses.

Reply:
https://github.com/davidpdrsn/diesel-factories/issues/9